src/ksyms.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+class Ksyms {
+public:
+  std::string resolve(uint64_t addr, bool show_offset);
+  ~Ksyms();
```

Comment:
For completeness, you'll want to delete the copy constructor and copy assignment operator. See rule of 3: https://en.cppreference.com/w/cpp/language/rule_of_three . Otherwise potential double free if instance is copied.